lib/smart_proxy_ipam/phpipam/phpipam_client.rb

```diff
@@ -164,28 +165,42 @@ def start_cleanup_task
     # @@ip_cache structure
     # {  
     #   "100.55.55.0/24":{  
-    #      "00:0a:95:9d:68:10": "100.55.55.1"
+    #      "00:0a:95:9d:68:10": {"ip": "100.55.55.1", "timestamp": "2019-09-17 12:03:43 -D400"}
     #   },
     #   "123.11.33.0/24":{  
-    #     "00:0a:95:9d:68:33": "123.11.33.1",
-    #     "00:0a:95:9d:68:34": "123.11.33.2",
-    #     "00:0a:95:9d:68:35": "123.11.33.3"
+    #      "00:0a:95:9d:68:33": {"ip": "123.11.33.1", "timestamp": "2019-09-17 12:04:43 -0400"}
+    #      "00:0a:95:9d:68:34": {"ip": "123.11.33.2", "timestamp": "2019-09-17 12:05:48 -0400"}
+    #      "00:0a:95:9d:68:35": {"ip": "123.11.33.3", "timestamp:: "2019-09-17 12:06:50 -0400"}
     #   }
     # }
     def init_cache
       logger.debug("Clearing ip cache.")
       @@m.synchronize do
-        @@ip_cache = {}
+        if @@ip_cache and not @@ip_cache.empty?
+          @@ip_cache.each do |key, values|
+            values.each do |mac, value|
+              if Time.now - Time.parse(value["timestamp".to_sym]) > DEFAULT_CLEANUP_INTERVAL
```

Comment:
You can use a symbol here `:timestamp` instead of converting a string to symbol.`"timestamp".to_sym`. Same goes for "ip".to_sym